anonymize_public_ips.py

```diff
@@ -0,0 +1,111 @@
+"""Helper script to check and clean IP Addresses in test data."""
+# pylint: disable=too-many-branches
+import re
+import os
+import sys
+import argparse
+import fileinput
+
+TEST_DATA_PATH = os.path.join("tests", "unit", "data")
+REPLACE_TEXT_IPV4 = "192.0.2.1"
+REPLACE_TEXT_IPV6 = "2001:DB8::1"
+
+# Patterns that could match IP address regex and are false positives
+PATTERNS_TO_SKIP = ("PRODID:Data::ICal",)
+
+# Reference: https://gist.github.com/dfee/6ed3a4b05cfe7a6faf40a2102408d5d8
+IPV4SEG = r"(?:25[0-5]|(?:2[0-4]|1{0,1}[0-9]){0,1}[0-9])"
+IPV4_ADDR_REGEX = r"(?:(?:" + IPV4SEG + r"\.){3,3}" + IPV4SEG + r")"
+IPV6SEG = r"(?:(?:[0-9a-fA-F]){1,4})"
+IPV6GROUPS = (
+    r"(?:" + IPV6SEG + r":){7,7}" + IPV6SEG,  # 1:2:3:4:5:6:7:8
+    r"(?:" + IPV6SEG + r":){1,7}:",  # 1::                                 1:2:3:4:5:6:7::
+    r"(?:" + IPV6SEG + r":){1,6}:" + IPV6SEG,  # 1::8               1:2:3:4:5:6::8   1:2:3:4:5:6::8
+    r"(?:" + IPV6SEG + r":){1,5}(?::" + IPV6SEG + r"){1,2}",  # 1::7:8             1:2:3:4:5::7:8   1:2:3:4:5::8
+    r"(?:" + IPV6SEG + r":){1,4}(?::" + IPV6SEG + r"){1,3}",  # 1::6:7:8           1:2:3:4::6:7:8   1:2:3:4::8
+    r"(?:" + IPV6SEG + r":){1,3}(?::" + IPV6SEG + r"){1,4}",  # 1::5:6:7:8         1:2:3::5:6:7:8   1:2:3::8
+    r"(?:" + IPV6SEG + r":){1,2}(?::" + IPV6SEG + r"){1,5}",  # 1::4:5:6:7:8       1:2::4:5:6:7:8   1:2::8
+    IPV6SEG + r":(?:(?::" + IPV6SEG + r"){1,6})",  # 1::3:4:5:6:7:8     1::3:4:5:6:7:8   1::8
+    r":(?:(?::" + IPV6SEG + r"){1,7}|:)",  # ::2:3:4:5:6:7:8    ::2:3:4:5:6:7:8  ::8       ::
+    r"fe80:(?::"
+    + IPV6SEG
+    + r"){0,4}%[0-9a-zA-Z]{1,}",  # fe80::7:8%eth0     fe80::7:8%1  (link-local IPv6 addresses with zone index)
+    r"::(?:ffff(?::0{1,4}){0,1}:){0,1}[^\s:]"
+    + IPV4_ADDR_REGEX,  # ::255.255.255.255  ::ffff:255.255.255.255  ::ffff:0:255.255.255.255 (IPv4-mapped IPv6 addresses and IPv4-translated addresses)
+    r"(?:"
+    + IPV6SEG
+    + r":){1,4}:[^\s:]"
+    + IPV4_ADDR_REGEX,  # 2001:db8:3:4::192.0.2.33  64:ff9b::192.0.2.33 (IPv4-Embedded IPv6 Address)
+)
+IPV6_ADDR_REGEX = "|".join([f"(?:{g})" for g in IPV6GROUPS[::-1]])  # Reverse rows for greedy match
+
+
+def replace(filename, search_exp, replace_exp):
+    """Replace line when a substitution is needed."""
+    report = ""
+    try:
+        for line in fileinput.input(filename, inplace=True):
+            if any(pattern_to_skip in line for pattern_to_skip in PATTERNS_TO_SKIP):
```

Comment:
a bit surprising to me that `PATTERNS_TO_SKIP` is a list of exact substring matches, rather than a list of regex patterns.

Reply:
it could be changed to regex if at some point is needed. For the current need, he simple substring matching was good enough